homeassistant/components/thinkingcleaner/switch.py

```diff
@@ -1,34 +1,47 @@
 """Support for ThinkingCleaner switches."""
-from datetime import timedelta
-import logging
 import time
-
-from pythinkingcleaner import Discovery
+import logging
+from datetime import timedelta
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
 
 from homeassistant import util
-from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.const import (STATE_ON, STATE_OFF)
 from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.const import CONF_HOST
 
 _LOGGER = logging.getLogger(__name__)
 
+REQUIREMENTS = ['pythinkingcleaner==0.0.3']
+
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(milliseconds=100)
 
 MIN_TIME_TO_WAIT = timedelta(seconds=5)
 MIN_TIME_TO_LOCK_UPDATE = 5
 
 SWITCH_TYPES = {
-    "clean": ["Clean", None, None],
-    "dock": ["Dock", None, None],
-    "find": ["Find", None, None],
+    'clean': ['Clean', None, None],
+    'dock': ['Dock', None, None],
+    'find': ['Find', None, None],
 }
 
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST): cv.string,
+})
+
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the ThinkingCleaner platform."""
+    from pythinkingcleaner import ThinkingCleaner, Discovery
 
-    discovery = Discovery()
-    devices = discovery.discover()
+    if config.get(CONF_HOST) is None:
+        discovery = Discovery()
+        devices = discovery.discover()
+    else:
+        host = config.get(CONF_HOST)
+        devices = [ThinkingCleaner(host, 'unknown')]
```

Comment:
I would suggest to switch the if and else for readability:
```suggestion
    host = config.get(CONF_HOST)
    if host:
        devices = [ThinkingCleaner(host, 'unknown')]
    else:
        discovery = Discovery()
        devices = discovery.discover()
```